### ISSUE TITLE
Selectively building workflows results in failing dependency resolution

### ISSUE THREAD

**Issue by tverbeiren** [viash-io/viash#513]
Using the (still experimental) dependency resolution feature of Viash to _build_ Nextflow workflows only works when all dependant components are built at the same time.

For example: specifying `viash ns build -q src/workflows` on one of our workflows results in

```
// dependency 'qc/fastqc' not found!
```

rather than

```
include { fastqc } from "$projectDir/../../../nextflow/qc/fastqc/main.nf"
```

**Comment by rcannood** [comment]
Good point!

So what needs to happen:

1. When a dependency could not be found, viash should consider this a build error (similar to missing resources)
2. The query parameters should only affect which components get built, but should not affect whether a local dependency can be found or not

**Comment by Grifs** [comment]
Fixed in Viash 0.8.0-RC2 and in the planned bigger release after that it is solved in a more structural way
